dev-system/dev/user.clj

```diff
@@ -35,6 +35,66 @@
           (fn [app-map]
             (u/map-values #(assoc-in % [:log :level] level) app-map))))
 
+(defn- maybe-set-all-modes
+  "A utility function for (conditionally) setting all the values of a map to
+  the same value."
+  [all new-modes]
```

Comment:
This docstring and the arguments are a bit confusing to me. It makes it sound like the second argument will get the 'all' value - there is no mention of the side-effect of looking up run-modes.